src/mscorlib/src/System/Reflection/AssemblyNameHelpers.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*============================================================
+**
+  Type:  AssemblyNameHelpers
+**
+==============================================================*/
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    internal static class AssemblyNameHelpers
+    {
+        private const int PUBLIC_KEY_TOKEN_LEN = 8;
+
+        public static String ComputeDisplayName(AssemblyName a)
+        {
+            if (a.Name == String.Empty)
+                throw new FileLoadException();
+
+            StringBuilder sb = new StringBuilder();
+            if (a.Name != null)
+            {
+                sb.AppendQuoted(a.Name);
+            }
+
+            if (a.Version != null)
+            {
+                Version canonicalizedVersion = a.Version.CanonicalizeVersion();
+                if (canonicalizedVersion.Major != ushort.MaxValue)
+                {
+                    sb.Append(", Version=");
+                    sb.Append(canonicalizedVersion.Major);
+                    sb.Append('.');
+                    sb.Append(canonicalizedVersion.Minor);
+                    sb.Append('.');
+                    sb.Append(canonicalizedVersion.Build);
+                    if (canonicalizedVersion.Revision != ushort.MaxValue)
+                    {
+                        sb.Append('.');
+                        sb.Append(canonicalizedVersion.Revision);
+                    }
+                }
+            }
+
+            String cultureName = a.CultureName;
+            if (cultureName != null)
+            {
+                if (cultureName == String.Empty)
+                    cultureName = "neutral";
+                sb.Append(", Culture=");
+                sb.AppendQuoted(cultureName);
+            }
+
+            byte[] pkt = null;
+            if (0 != (a.Flags & AssemblyNameFlags.PublicKey))
+                pkt = a.GetPublicKeyToken();
+
+            if(pkt != null)
+            {
+                if (pkt.Length > PUBLIC_KEY_TOKEN_LEN)
+                    throw new ArgumentException();
+
+                sb.Append(", PublicKeyToken=");
+                if (pkt.Length == 0)
+                    sb.Append("null");
+                else
+                {
+                    foreach (byte b in pkt)
+                    {
+                        sb.Append(b.ToString("x2", CultureInfo.InvariantCulture));
+                    }
+                }
+            }
+
+            if (0 != (a.Flags & AssemblyNameFlags.Retargetable))
+                sb.Append(", Retargetable=Yes");
+
+            AssemblyContentType contentType = a.Flags.ExtractAssemblyContentType();
```

Comment:
Careful now: RuntimeAssemblyName.Flags and AssemblyName.Flags don't return the same value. In particular, AssemblyName.Flags don't include the content type. You have to get that from AssemblyName.ContentType.